### ISSUE TITLE
Short notation?

### ISSUE THREAD

**Issue by kostya** [ysbaddaden/shards#75]
```
 dependencies:
   cron_scheduler:
     github: kostya/cron_scheduler
   nagios_check:
     github: kostya/nagios_check
   pg:
     github: will/crystal-pg
  redis:
    github: stefanwille/crystal-redis
```
this is too noisy to manual edit, repeated some parts

why not add default short notation together with full notation, which will be used in most cases:

`[git|github|bitbucket|path]:user/repo[#version|branch|tag][@revision]`

```
 dependencies:
   - github:kostya/cron_scheduler@bde084a30
```

or even group by github:

```
 dependencies:
   github:
     - kostya/cron_scheduler
     - kostya/nagios_check
     - will/crystal-pg
     - stefanwille/crystal-redis
```

**Comment by asterite** [comment]
I personally wouldn't make simple things complex. Editing a `shard.yml` is probably %0.01 of the time compared to coding the app.

**Comment by ysbaddaden** [comment]
I was about to say the same.

I admit the format is verbose, but it makes things explicit. I skip a line between each dependency so it's nicer to skim throught.

I agree there is the repetition of the shard name; maybe an array would have been enough, but then we'd lose the name in favor to the github user/repo which is more verbose and often not the actual name, at least for now, eg: `crystal-pg`, `minitest.cr`, ...

Thought I need to detect and print an error when the specified name doesn't match the shard name, to fix #72
